version: '3.8'

services:
  # PostgreSQL Datenbank
  db:
    image: postgres:15
    container_name: odoo_db
    environment:
      POSTGRES_DB: postgres
      POSTGRES_USER: odoo
      POSTGRES_PASSWORD: odoo
      PGDATA: /var/lib/postgresql/data/pgdata
    #volumes:
      #- odoo-db-data:/var/lib/postgresql/data/pgdata
    ports:
      - "5432:5432"
    restart: unless-stopped
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U odoo -d postgres"]
      interval: 10s
      timeout: 5s
      retries: 10

  # Odoo Application
  odoo:
    image: odoo:17
    container_name: odoo_app
    depends_on:
      db:
        condition: service_healthy
    environment:
      HOST: db
      USER: odoo
      PASSWORD: odoo
    ports:
      - "8069:8069"
    volumes:
      #- odoo-web-data:/var/lib/odoo
      - ./config:/etc/odoo
    restart: unless-stopped
    command: --
      --database=odoo
      --db_host=db
      --db_port=5432
      --db_user=odoo
      --db_password=odoo
      --addons-path=/usr/lib/python3/dist-packages/odoo/addons
      --xmlrpc-interface=0.0.0.0
      --xmlrpc-port=8069
      --without-demo=False
      --init=base,sale,purchase

volumes:
  odoo-web-data:
  odoo-db-data:

networks:
  default:
    name: odoo-network